internal/webhook/external/providers/github/provider.go

```diff
@@ -45,66 +42,22 @@ func (p *provider) Name() string {
 }
 
 func (p *provider) Authenticate(r *http.Request) error {
-	sig := r.Header.Get("X-Hub-Signature-256")
-	if sig == "" {
-		return ErrMissingSignature
-	}
-
-	if !strings.HasPrefix(sig, "sha256=") {
-		return ErrInvalidSignature
-	}
-
-	b, err := xhttp.PeakBody(r)
+	payload, err := gh.ValidatePayload(r, []byte(p.secret))
```

Comment:
Internally, this doesn't use an `io.LimitReader`, so we should read the payload/body more carefully ourselves, then use `gh.ValidateSignature()`. If you can't find that function, it's probably because of [this](https://github.com/akuity/kargo/pull/4000/files/1bbd3ce92315c82fdd2e53c67aa475c5628e32ef..a58e9d4c5afd627801380816692583cc2ddcd67c#r2078237974).

Reply:
Nevermind... I was somehow looking at out of date diffs when I wrote this.